src/test/java/com/wikia/webdriver/testcases/discussions/ReportingPostTests.java

```diff
@@ -0,0 +1,620 @@
+  }
+
+  // Discussions moderator on desktop
+
+  @Test(groups = "discussions-loggedInDiscussionsModeratorDesktopReporting",
+      dependsOnMethods = {
+          "userOnDesktopCanReportPostOnPostDetailsPage",
+          "anonUserOnDesktopCanNotSeeReportedPostOnPostDetailsPage",
+          "userOnDesktopCannotSeeReportedIndicatorOnPostsReportedByAnotherUserOnPostDetailsPageAndCanReportThatPost"})
+  @Execute(asUser = User.DISCUSSIONS_MODERATOR)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
+  public void moderatorOnDesktopCanApproveReportedPostOnPostDetailsPage() {
+    final String postId = testsPostData.get(POST_ON_DESKTOP_ON_POST_DETAILS).getId();
+
+    PostDetailsPage page = new PostDetailsPage().open(postId);
+
+    assertThatDiscussionsModeratorCanSeeAndApproveReportedPost(page, postId);
+  }
+
+  @Test(groups = "discussions-loggedInDiscussionsModeratorDesktopReporting",
+      dependsOnMethods = {
+          "userOnDesktopCanReportPostOnPostDetailsPage",
+          "anonUserOnDesktopCanNotSeeReportedPostOnPostDetailsPage",
+          "userOnDesktopCannotSeeReportedIndicatorOnPostsReportedByAnotherUserOnPostDetailsPageAndCanReportThatPost",
+          "moderatorOnDesktopCanApproveReportedPostOnPostDetailsPage",
+          "userOnDesktopCannotReReportPostOnPostDetailsPage",
+          "userOnDesktopCanReportApprovedPostOnPostDetailsPage"})
+  @Execute(asUser = User.DISCUSSIONS_MODERATOR)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
+  public void moderatorOnDesktopCanDeleteReportedPostOnPostDetailsPage() {
+    final String postId = testsPostData.get(POST_ON_DESKTOP_ON_POST_DETAILS).getId();
+
+    PostDetailsPage page = new PostDetailsPage().open(postId);
+
+    assertThatDiscussionsModeratorCanSeeAndRemoveReportedPost(page, postId);
+  }
+
+  private void assertThatReportPostOptionIsNotAvailableOn(final PageWithPosts page) {
+    assertThatReportPostOptionIsNotAvailableFor(page.getPost().findNewestPost());
+  }
+
+  private void assertThatNoPostHasReportedIndicator(final PageWithPosts pageWithPosts) {
+    Assertion.assertTrue(pageWithPosts.getPost().getReportedPosts().isEmpty(),
+        "Anonymous user should not see reported indicator.");
+  }
+
+  private PostEntity createAndGetNewPost(final PostsListPage postsListPage, final PostsCreator postsCreator) {
+    PostEntity.Data postData = postsCreator.click()
+        .closeGuidelinesMessage()
+        .addPostWithRandomData();
+
+    return postsListPage.getPost()
+        .waitForPostToAppearWith(postData.getDescription())
+        .findNewestPost();
+  }
+
+  private PostEntity addPostAndGoToPostDetailsPage(final PostsListPage postsListPage, final PostsCreator postsCreator) {
+    createAndGetNewPost(postsListPage, postsCreator).click();
+
+    new Transitions(driver).waitForPostDetailsPageTransition();
+
+    return new PostDetailsPage().getPost().findNewestPost();
+  }
+
```

Comment:
get rid of all this assert methods, just us for example:
`private boolean CanPostBeReported` and assert on that